Pod/Classes/BONTag.m

```diff
@@ -0,0 +1,235 @@
+
+    return [self initWithStartTag:startTag endTag:endTag escapeString:(NSString *)kBONTagDefaultEscapeString textable:textable];
+}
+
+- (id)copyWithZone:(NSZone *)zone
+{
+    __typeof(self) tag = [[self.class alloc] init];
+
+    tag.startTag = self.startTag;
+    tag.endTag = self.endTag;
+    tag.escapeString = self.escapeString;
+    tag.textable = self.textable;
+    tag.ranges = self.ranges;
+
+    return tag;
+}
+
+- (NSString *)description
+{
+    NSString *description = [NSString stringWithFormat:@"<%@: %p, startTag: %@, endTag: %@, textable: <%@: %p>>", NSStringFromClass(self.class), self, self.startTag, self.endTag, NSStringFromClass(self.textable.class), self.textable];
+
+    return description;
+}
+
+#pragma mark - Tag Matching
+
++ (NSArray *)escapedRangesInString:(NSString **)string withTags:(NSArray *)tags stripEscapeCharacters:(BOOL)stripEscapeCharacters
+{
+    NSMutableArray *escapedRanges = [NSMutableArray array];
+
+    NSString *theString = *string;
+
+    NSRange searchRange = NSMakeRange(0, theString.length);
+
+    // Iterate over the string finding each escape in order until there are no more escape strings
+    while (YES) {
+        BONTag *nextTag;
+        NSRange nextEscapeRange;
+        NSRange nextEscapedTagRange;
+
+        // Find the next escape string
+        for (BONTag *tag in tags) {
+            NSRange escapeRange = [theString rangeOfString:tag.escapeString options:0 range:searchRange];
+
+            if (escapeRange.location != NSNotFound) {
+                if (!nextTag || (escapeRange.location < nextEscapeRange.location)) {
+                    // Check if this character is escaping a start or end tag
+                    for (NSString *tagString in @[ tag.startTag, tag.endTag ]) {
+                        // Check if there is room for this tag to exist after the escape character
+                        if ((NSInteger)(NSMaxRange(escapeRange) + tagString.length) < theString.length) {
+                            // Check if the following characters are actually the tag
+                            NSRange potentialEscapedTagRange = NSMakeRange(NSMaxRange(escapeRange), tagString.length);
+                            if ([[theString substringWithRange:potentialEscapedTagRange] isEqualToString:tagString]) {
+                                nextTag = tag;
+                                nextEscapeRange = escapeRange;
+                                nextEscapedTagRange = potentialEscapedTagRange;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        if (!nextTag) {
+            break;
+        }
+
+        NSRange range = NSMakeRange(nextEscapeRange.location, nextEscapedTagRange.length + nextEscapeRange.length);
+
+        if (stripEscapeCharacters) {
+            theString = [theString stringByReplacingOccurrencesOfString:nextTag.escapeString withString:@"" options:0 range:nextEscapeRange];
```

Comment:
#nth I believe this will be faster if you create a mutable string, mutate it a bunch inside the loop, instead of copying the string each time.